tools/lib/provision_inner.py

```diff
@@ -211,7 +211,11 @@ def main(options: argparse.Namespace) -> int:
             print("No need to run `scripts/setup/configure-rabbitmq.")
 
         dev_template_db_status = DEV_DATABASE.template_status()
-        if options.is_force or dev_template_db_status == 'needs_rebuild':
+        if 'CIRCLECI' in os.environ:
```

Comment:
ok, we discussed on chat passing in a more abstract flag from the "outer" provision (something like SKIP_DEV_DB)

Reply:
Updated! (CI is flake left as it is for debugging purposes.)

Reply:
@timabbott Do we ever call `tools/lib/provision_inner.py` from an actual command line?  It feels clunky to have command-line arguments instead of just using environment variables (especially for setting the `is_production_test_suite` flag).  I guess that's a bit beyond the scope of this PR to change our approach there.

@amanagr When I suggested `SKIP_DEV_DB` I was actually thinking of an env var, not a command flag--that's why I'm asking Tim about this.

Reply:
We do not, but I'm generally not a fan of using environment variables as an interface unless we absolutely have to, because it reduces predictability.